apps/src/music/views/SoundsPanel.tsx

```diff
@@ -158,50 +195,135 @@ const SoundsPanel: React.FunctionComponent<SoundsPanelProps> = ({
   onPreview,
 }) => {
   const folders = library.getAllowedSounds(undefined);
-  const libraryGroupPath = library.groups[0].path;
-
-  // Generate a flat list of entries to render.  We need a flat list because
-  // we will make the headers sticky.
-  const entries: {
-    type: 'folder' | 'sound';
-    folder: SoundFolder;
-    sound?: SoundData;
-  }[] = [];
-  folders.forEach(folder => {
-    entries.push({type: 'folder', folder: folder});
-    folder.sounds.forEach(sound => {
-      entries.push({type: 'sound', folder: folder, sound: sound});
+  const libraryGroupPath = library.libraryJson.path;
+
+  const [selectedFolder, setSelectedFolder] = useState<SoundFolder>(
+    library.getFolderForSoundId(currentValue) || folders[0]
+  );
+  const [mode, setMode] = useState<string>('packs');
+  const [filter, setFilter] = useState<string>('all');
+
+  const currentFolderRef: React.MutableRefObject<HTMLDivElement | null> =
+    useRef(null);
+  const currentSoundRef: React.MutableRefObject<HTMLDivElement | null> =
+    useRef(null);
+
+  const onModeChange = (value: string) => {
+    setMode(value);
+  };
+
+  const onFilterChange = (value: string) => {
+    setFilter(value);
+  };
+
+  useEffect(() => {
+    // This timeout allows the scrolling to work when wrapping the content with FocusLock.
+    setTimeout(() => {
+      currentFolderRef.current?.scrollIntoView();
+      currentSoundRef.current?.scrollIntoView();
+    }, 0);
+  }, []);
+
+  const currentFolderRefCallback = (ref: HTMLDivElement) => {
+    currentFolderRef.current = ref;
+  };
+
+  const currentSoundRefCallback = (ref: HTMLDivElement) => {
+    currentSoundRef.current = ref;
+  };
+
+  let possibleSounds: SoundData[] = [];
+  let rightColumnSounds: SoundData[] = [];
+
+  if (mode === 'packs') {
+    possibleSounds = selectedFolder.sounds;
+  } else {
+    folders.forEach(folder => {
+      folder.sounds.forEach(sound => {
+        possibleSounds.push(sound);
+      });
     });
-  });
+  }
+
+  if (filter === 'all') {
+    rightColumnSounds = possibleSounds;
+  } else {
+    rightColumnSounds = possibleSounds.filter(
+      sound => filter === 'all' || sound.type === filter
```

Comment:
nit: can remove the `filter === 'all'` here, right?

Reply:
Good catch.